lib/Cake/Network/CakeSocket.php

```diff
@@ -129,8 +129,12 @@ public function connect() {
 			$this->disconnect();
 		}
 
+		$hasProtocol = strpos($this->config['host'], '://') !== false;
```

Comment:
You will have to ask @markstory .
I could think of performance reasons, even though it would be very minor - and only relevant for a lot of loop calls.